### ISSUE TITLE
Support multiple Gateway Listener certificateRefs

### ISSUE THREAD

**Issue by rainest** [Kong/kubernetes-ingress-controller#2604]
Per https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1alpha2.GatewayTLSConfig

> A single CertificateRef to a Kubernetes Secret has “Core” support. Implementations MAY choose to support attaching multiple certificates to a Listener, but this behavior is implementation-specific.

#2580 only implements the core requirement, and rejects configurations with multiple certificates.

Kong certificate entities can hold two certificates, and will select the served certificate based on client algorithm support (ECDSA or RSA): https://docs.konghq.com/gateway/latest/admin-api/#add-certificate

To support this for Gateway APIs, we would need to:

- [ ] Increase the certificateRef limit to 2.
- [ ] If two certificates are present, verify that they use different signature algorithms (not 100% sure about this--double-check with the Kong gateway team to see if there are other reasons to upload multiple).
- [ ] For Listeners with 2 certificateRefs, create a Kong certificate with one certificate and key in `cert` and `key` and the other in `cert_alt` and `key_alt`.

**Comment by scseanchow** [comment]
@rainest is this extended support, does it make sense for us to push this out to a further milestone?

**Comment by rainest** [comment]
Yes, per the spec this is extended. Core only requires supporting a single certificate. The main reason for us to add it is because there's an existing Kong feature we can take advantage of with multiple.
